pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -313,11 +315,32 @@ def adjoint_diff_support_check(self, tape):
                     raise QuantumFunctionError(
                         "Lightning adjoint differentiation method does not currently support the Hermitian observable"
                     )
+                if isinstance(m.obs, Hamiltonian):
+                    raise QuantumFunctionError(
+                        "Adjoint differentiation method does not currently support the Hamiltonian observable"
+                    )
+                if isinstance(m.obs, SparseHamiltonian):
+                    raise QuantumFunctionError(
+                        "Adjoint differentiation method does not currently support the SparseHamiltonian observable"
+                    )
+                if isinstance(m.obs, Hermitian):
+                    raise QuantumFunctionError(
+                        "Lightning adjoint differentiation method does not currently support the SparseHamiltonian observable"
+                    )
+
```

Comment:
Rather than duplicating the if statements, we can make a single if statement and check all unsupported types, and use f-string "{obs.name}" to grab the type name for the error.

Reply:
forgot to push, sry